src/drivers/intel/dmic.c

```diff
@@ -1292,7 +1292,7 @@ static int dmic_set_config(struct dai *dai, struct ipc_config_dai *common_config
 static void dmic_start(struct dai *dai)
 {
 	struct sof_ipc_dai_dmic_params **uncached_dmic_prm = cache_to_uncache(&dmic_prm[0]);
-	int *uncached_dmic_active_fifos = cache_to_uncache(&dmic_active_fifos);
+	int *uncached_dmic_active_fifos_mask = cache_to_uncache(&dmic_active_fifos_mask);
```

Comment:
@lyakh @lgirdwood Hmm, this is not nice. On XTOS, data declared as shared is still mapped to cached addresses, so the above call is needed. On Zephyr, this are already uncached so above code is not needed. Having application code to do both seems very non-ideal and I can imagine we can easily have subtle bugs that only happen on one OS.

@singalsu Code is correct, just making a note.

Reply:
Yep, it's not nice looking code at all. Also the dmic_remove() dmic_prm[] set to NULL looks rather confusing.

Reply:
@singalsu you need to go over the private structures/member used by DMIC and decide what is shared between cores and what is a single core only. Converting on the fly like this is not guaranteed to work since the HW works on cache line alignment and cache line size (not variable pointer and variable alignment).

 All the shared data must then be allocated like so
```c
struct private_nonshared1 {
    blah;
};

struct private_shared1 {
};

struct private_shared2 {
};

struct dmic_data {
    struct private_nonshared1 *nonshared1; // cached alias
    struct private_shared1 *shared; // uncached alias
    struct private_shared2 *shared; // uncached alias
};
```